### DISCUSSION TITLE
A canonical multi-stage Dockerfile

### DISCUSSION THREAD

**Discussion by meglio** [pnpm/pnpm#3644] [Q&A]
Is there a canonical example of a multi-stage node-based Dockerfile that uses `pnpm`? I could not find any in your documentation.

As an example, this (Next.js Dockerfile example](https://nextjs.org/docs/deployment#docker-image) does not work if I change `npm` to `pnpm` and adjust the argument names accordingly.

Would be nice to have an "official" working example of a multi-stage Dockerfile that uses `pnpm` instead of npm or yarn.

Please advise. -- thanks, Anton

**Comment by zkochan** [comment]
Currently, the recommended way is to use the [standalone script](https://pnpm.io/installation#using-a-standalone-script) to install pnpm in a docker image. And Node.js should be installed before that.

But we are also working on a standalone version of pnpm, so you'll be able to install pnpm without Node.js. And we plan to add support for system specific package managers, see: https://github.com/pnpm/pnpm/issues/3633

**Comment by ghost** [comment]
https://github.com/pnpm/pnpm/issues/3114
